### PR TITLE
[BUG] - Improved error handling to avoid nil ptrs

### PR DESCRIPTION
**Pull request by komer3**
**What type of PR is this?**

**What this PR does / why we need it**:
BUG FIX: Using the correct method to unwrap errors from linodego and handling it

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: controller/linodecluster_controller.go [hunk 1]
```diff
@@ -165,7 +164,12 @@ func (r *LinodeClusterReconciler) reconcileCreate(ctx context.Context, logger lo
 	linodeNB, err := services.CreateNodeBalancer(ctx, clusterScope, logger)
 	if err != nil {
 		setFailureReason(clusterScope, cerrs.CreateClusterError, err, r)
+		return err
+	}
 
+	if linodeNB == nil {
```
Comment (rahulait) [inline controller/linodecluster_controller.go:170]: I wonder if api will ever return nil nodebalancer and no error.

**Review by rahulait — DISMISSED** [review]
Except the minor comment/question, the PR looks good to me.

File: cloud/services/object_storage_buckets.go [hunk 2]
```diff
@@ -18,9 +17,11 @@ func EnsureObjectStorageBucket(ctx context.Context, bScope *scope.ObjectStorageB
 		bScope.Bucket.Spec.Cluster,
 		bScope.Bucket.Name,
 	)
-	linodeErr := &linodego.Error{}
-	if errors.As(err, linodeErr) && linodeErr.StatusCode() != http.StatusNotFound {
-		return nil, fmt.Errorf("failed to get bucket from cluster %s: %w", bScope.Bucket.Spec.Cluster, err)
+	if err != nil {
+		linodeErr := linodego.NewError(err)
+		if linodeErr.StatusCode() != http.StatusNotFound {
```
Comment (AshleyDumaine) [inline cloud/services/object_storage_buckets.go:22]: I think for this we should be able to use `util.IgnoreLinodeAPIError(err, http.StatusNotFound)`

Comment (komer3) [reply cloud/services/object_storage_buckets.go]: Use I was able to use that. Thanks for pointing that out!

**Review by rahulait — APPROVED** [review]

**Pull request merged**